src/components/settings/SettingsLMSTab/LMSConfigs/DegreedConfig.jsx

```diff
@@ -45,6 +46,12 @@ const DegreedConfig = ({ id, onClick }) => {
     }
   };
 
+  const validateUrl = (input) => {
```

Comment:
since this idea of the urlValid, setUrlValid is common across all the various config JSX files, fun idea if you are interested (optional) is to abstract out the common idea of a valid url, you could have sth like this 

so for ex

```src/components/settings/SettingsLMSTab/hooks.jsx```

could export a function like this that exports this state, 

```
function useValidatedURL(inputURL) {
    const [urlValid, setUrlValid] = React.useState(false);
    const isURLValid = useMemo(() => {
        const urlBool = (urlValidation(input) || input.length === 0);
        setUrlValid(urlBool);
        return isURLValid; // you may not need this return val anywhere so you can skip if you like
    }, inputURL); // only re evals setUrlValid if the inputURL is different from before
    return [urlValid, setUrlVal];
}
```

then in all usage code you could use

```
const [urlValid, setUrlValid] = useValidatedURL(degreedBaseUrl);
const validateUrl = input => { setDegreedBaseUrl(input); /* now urlValid should automatically update when degreedBaseURL changes */ }
```

not much of a refactor but I think it lends itself to a bit cleaner setup if you want to explore the option

the one future advantage is if we change the way we manage that state (such as calling other hooks as side effects), a change only needs to be made inside the use hook